src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -183,6 +185,37 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return $env;
         }
 
+        if ('url' === $prefix) {
+            $parsedEnv = parse_url($env);
+
+            if (false === $parsedEnv) {
+                throw new RuntimeException(sprintf('Invalid URL in env var "%s"', $name));
+            }
+            if (!isset($parsedEnv['scheme'], $parsedEnv['host'])) {
+                throw new RuntimeException(sprintf('Invalid URL env var "%s": schema and host expected, %s given.', $name, $env));
+            }
+            $parsedEnv += [
+                'port' => null,
+                'user' => null,
+                'pass' => null,
+                'path' => null,
+                'query' => null,
+                'fragment' => null,
+            ];
+
+            // remove the '/' separator
+            $parsedEnv['path'] = '/' === $parsedEnv['path'] ? null : substr($parsedEnv['path'], 1);
+
+            return $parsedEnv;
+        }
+
+        if ('query_string' === $prefix) {
+            $queryString = parse_url($env, PHP_URL_QUERY) ?? $env;
```

Comment:
Should use `?:` I think as parse_url can return false also

Reply:
> If the component parameter is specified, parse_url() returns a string (or an integer, in the case of PHP_URL_PORT) instead of an array. If the requested component doesn't exist within the given URL, NULL will be returned.

:crossed_fingers:  but i didnt spot a case which returns false when a component is set.

Reply:
`var_export(parse_url(':', PHP_URL_QUERY));` displays false :)

Reply:
:sob: